src/components/App.tsx

```diff
@@ -382,21 +401,67 @@ export class App extends React.Component<any, AppState> {
         });
       });
       this.socket.on("new-user", async (socketID: string) => {
-        this.broadcastSocketData({
-          type: "SCENE_UPDATE",
-          payload: {
-            elements: elements.filter(element => {
-              return element.id !== this.state.editingElement?.id;
-            }),
-            appState: this.state,
-          },
-        });
+        this.broadcastSocketData(
+          this.getSocketData({
+            type: "SCENE_UPDATE",
+          }),
+        );
       });
     }
   };
 
-  private broadcastSocketData = async (data: SocketUpdateData) => {
+  // brands socketData to ensure we're using the getSocketData() helper
+  private _getSocketData<T>(data: T): T & { _brand: "socketData" } {
+    return data as typeof data & { _brand: "socketData" };
+  }
+
+  private getSocketData(
+    data: SocketUpdateDataSource[keyof SocketUpdateDataSource],
```

Comment:
I'm not a fan of this convoluted API.

What about making two functions:

```js
this.broadcastMouseLocation(pointerCoords);
this.broadcastSceneUpdate();
```

Reply:
We can even do what `getSocketData` does inside `broadcastSocketData` and not split anything. But the idea was to ensure we process the data correctly (hence the brand). If we call the `broadcastSocketData()` directly, we won't be able to ensure that. But I think it's not that important right now. Will change.